rdkit/Chem/Pharm3D/Pharmacophore.py

```diff
@@ -58,74 +59,72 @@ def getFeature(self, i):
     return self._feats[i]
 
   def getUpperBound(self, i, j):
-    if (i > j):
+    if i > j:
       j, i = i, j
     return self._boundsMat[i, j]
 
   def getLowerBound(self, i, j):
-    if (j > i):
+    if j > i:
       j, i = i, j
     return self._boundsMat[i, j]
 
   def _checkBounds(self, i, j):
     " raises ValueError on failure "
     nf = len(self._feats)
-    if (i < 0) or (i >= nf):
-      raise ValueError("Index out of bound")
-    if (j < 0) or (j >= nf):
-      raise ValueError("Index out of bound")
-    return True
+    if (0 <= i < nf) and (0 <= j < nf):
+      return True
+    raise ValueError("Index out of bound")
 
   def setUpperBound(self, i, j, val, checkBounds=False):
-    if (checkBounds):
+    if checkBounds:
       self._checkBounds(i, j)
-    if (i > j):
+    if i > j:
       j, i = i, j
     self._boundsMat[i, j] = val
 
   def setLowerBound(self, i, j, val, checkBounds=False):
-    if (checkBounds):
+    if checkBounds:
       self._checkBounds(i, j)
-    if (j > i):
+    if j > i:
       j, i = i, j
     self._boundsMat[i, j] = val
 
   def getUpperBound2D(self, i, j):
-    if (i > j):
+    if i > j:
       j, i = i, j
     return self._boundsMat2D[i, j]
 
   def getLowerBound2D(self, i, j):
-    if (j > i):
+    if j > i:
       j, i = i, j
     return self._boundsMat2D[i, j]
 
   def setUpperBound2D(self, i, j, val, checkBounds=False):
-    if (checkBounds):
+    if checkBounds:
       self._checkBounds(i, j)
-    if (i > j):
+    if i > j:
       j, i = i, j
     self._boundsMat2D[i, j] = val
 
   def setLowerBound2D(self, i, j, val, checkBounds=False):
-    if (checkBounds):
+    if checkBounds:
       self._checkBounds(i, j)
-    if (j > i):
+    if j > i:
       j, i = i, j
     self._boundsMat2D[i, j] = val
 
   def __str__(self):
-    res = ' ' * 13
+    res = ' ' * 14
     for i, iFeat in enumerate(self._feats):
-      res += '% 12s ' % iFeat.GetFamily()
+      res += '%12s ' % iFeat.GetFamily()
     res += '\n'
     for i, iFeat in enumerate(self._feats):
-      res += '% 12s ' % iFeat.GetFamily()
-      for j, jFeat in enumerate(self._feats):
+      res += '%13s ' % iFeat.GetFamily()
+      for j, _ in enumerate(self._feats):
         if j < i:
-          res += '% 12.3f ' % self.getLowerBound(i, j)
+          res += '%12.3f ' % self.getLowerBound(i, j)
```

Comment:
This generates different output: it's no longer padded to always be 12 characters long, which will screw up the formatting:
```
In [6]: '>'+'%4.2f'%6.12789+'<'
Out[6]: '>6.13<'

In [7]: '>'+'% 4.2f'%6.12789+'<'
Out[7]: '> 6.13<'
```